frontend/packages/dev-console/src/utils/pipeline-actions.tsx

```diff
@@ -131,13 +131,36 @@ export const startPipeline: KebabAction = (
   },
 });
 
-export const rerunPipeline: KebabAction = (kind: K8sKind, pipelineRun: PipelineRun) => {
-  // The returned function will be called using the 'kind' and 'obj' in Kebab Actions
+type RerunPipelineData = {
+  pipelineRunRedirect?: boolean;
+};
+export const rerunPipeline: KebabAction = (
+  kind: K8sKind,
+  pipelineRun: PipelineRun,
+  resources: any,
+  customData: RerunPipelineData = {},
+) => {
+  const { pipelineRunRedirect = true } = customData;
+
+  const sharedProps = { label: 'Start Last Run', accessReview: {} };
+
+  if (
+    !pipelineRun ||
+    !_.has(pipelineRun, 'metadata.name') ||
+    !_.has(pipelineRun, 'metadata.namespace')
+  ) {
```

Comment:
Why are you checking for metadata.name and namespace here? If `pipelineRun` is there then metadata should be there.

Reply:
Not according to the type. `PipelineRun` doesn't define it, but extends `K8sResourceKind` and it specifies it as optional. Of which inside `metadata`, `name` and `namespace` are also optional.

Reply:
@sahil143 unless the type we are currently using says this, I won't write code that "assumes" optional params are there. Types should not be just ignored when it's inconvenient. This code is currently only executed on click so it's not a render-cycle slowdown.